src/main/radiusd.c

```diff
@@ -621,6 +625,12 @@ int main(int argc, char *argv[])
 	 */
 	state = fr_state_init(NULL);
 
+#ifdef HAVE_SYSTEMD
+	if (sd_notify(0, "READY=1") <=0) {
+		WARN("Something goes wrong on send READY event to systemd");
```

Comment:
Could you change that to:

```
ret = sd_notify(0, "READY=1");
if (ret < 0) WARN("Failed notifying systemd that process is READY: %s", fr_syserror(ret)));
```

You'll have to add an ```int ret``` somewhere too.  A return code of 0 means that $NOTIFY_SOCKET wasn't set, so that'd happen any time we were running in debug mode, so we don't want to emit warning messages there...

Reply:
@arr2036 what do you think to change the STATUS also?

e.g:

```
sd_notifyf(0, "READY=1\n"
        "STATUS=Processing requests…\n"
        "MAINPID=%lu",
        (unsigned long) getpid());
```

Reply:
Sure...

Reply:
therefore, it could be something like:

```
ret = sd_notifyf(0, "READY=1\n"
            "STATUS=Processing requests…\n"
            "MAINPID=%lu",
            (unsigned long) getpid());
if (ret < 0) {
    WARN("Failed notifying systemd that process is READY: %s", fr_syserror(ret)));
}
```

Reply:
Hi, just a summary of the last changes:
* avoid WARN if NOTIFY_SOCKET is not available
* add STATUS event
* avoid daemonizing in existing .service (missing in the last changes)

@jpereira I sent MAINPID only if I daemonized the process and not with READY event. I think that could be more correct. Let me know WDYT.